### PR TITLE
[RFR] Remove deprecated Link component

### PR DESCRIPTION
**Pull request by jaytula**
- Remove deprecated Link component
- Remove `WithStyles` wraps  `SingleFieldList.spec.js`
- Update two components to import from react-router-dom for `Link`: `ReferenceField` and `SingleFieldList`

### CODE REVIEW

**Comment by jaytula** [comment]
Sure, will do.

File: packages/ra-ui-materialui/src/index.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import Link from './Link';
+import { Link } from 'react-router-dom';
```
Comment (djhi) [inline packages/ra-ui-materialui/src/index.js]: I think we don't need to re-export the Link component. Ping @fzaninotto

Comment (fzaninotto) [reply packages/ra-ui-materialui/src/index.js]: if it makes the upgrade easier... Just mark it as deprecated down where you export it.

**Comment by fzaninotto** [comment]
I'm the one who deprecated Link (in #2050), but I'm afraid it breaks link styling. I'll have to checkout your branch and test.

**Comment by fzaninotto** [comment]
Indeed this breaks styling of ReferenceFields. It should not have been deprecated in the first place (my bad). So I'm closing this PR and removing the deprecated mention. But thanks a lot for your contribution anyway!

**Pull request closed without merging**